File: solutions-blind/javascript/007-caesar/solution.mjs
function shiftChar(ch, shift) {
  const code = ch.charCodeAt(0);
  if (code >= 97 && code <= 122) {
    return String.fromCharCode(97 + (((code - 97 + shift) % 26) + 26) % 26);
  }
  if (code >= 65 && code <= 90) {
    return String.fromCharCode(65 + (((code - 65 + shift) % 26) + 26) % 26);
  }
  return ch;
}

export function encrypt(text, shift) {
  let out = "";
  for (const ch of text) out += shiftChar(ch, shift);
  return out;
}

export function decrypt(text, shift) {
  return encrypt(text, -shift);
}

File: solutions-blind/python/007-caesar/solution.py
def _shift_char(ch: str, shift: int) -> str:
    if "a" <= ch <= "z":
        base = ord("a")
        return chr(base + (ord(ch) - base + shift) % 26)
    if "A" <= ch <= "Z":
        base = ord("A")
        return chr(base + (ord(ch) - base + shift) % 26)
    return ch


def encrypt(text: str, shift: int) -> str:
    return "".join(_shift_char(ch, shift) for ch in text)


def decrypt(text: str, shift: int) -> str:
    return encrypt(text, -shift)
